pkg/api/steve/collections/rbac.go

```diff
@@ -0,0 +1,315 @@
+package collections
+
+import (
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/rancher/pkg/controllers/managementagent/nslabels"
+	pkgrbac "github.com/rancher/rancher/pkg/rbac"
+	corecontrollers "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+	rbaccontrollers "github.com/rancher/wrangler/pkg/generated/controllers/rbac/v1"
+	"github.com/sirupsen/logrus"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	suffix                     = "-collections"
+	clusterRoleOwnerAnnotation = "authz.cluster.cattle.io/clusterrole-owner"
+)
+
+var ErrNotReady = errors.New("APIs have not been synced yet")
+
+type rbacHandler struct {
+	namespaceCache    corecontrollers.NamespaceCache
+	clusterRolesCache rbaccontrollers.ClusterRoleCache
+	rolesCache        rbaccontrollers.RoleCache
+	clusterRoles      rbaccontrollers.ClusterRoleClient
+	roles             rbaccontrollers.RoleClient
+	roleBindingsCache rbaccontrollers.RoleBindingCache
+	roleBindings      rbaccontrollers.RoleBindingController
+	apis              APIResourceWatcher
+}
+
+func (h *rbacHandler) OnChange(_ string, rb *rbacv1.RoleBinding) (*rbacv1.RoleBinding, error) {
+	if rb == nil {
+		return nil, nil
+	}
+	if strings.HasSuffix(rb.GetName(), suffix) { // this rolebinding was created by this controller, don't recurse
```

Comment:
Nit:
I'd rather we do this check with a label/annotation than based on naming syntax, especially since we intend this to be active (AFAICT) in downstream clusters.